core/server/models/user.js

```diff
@@ -554,6 +554,12 @@ User = ghostBookshelf.Model.extend({
                 return bcryptCompare(object.password, user.get('password')).then(function then(matched) {
                     if (!matched) {
                         return Promise.resolve(self.setWarning(user, {validate: false})).then(function then(remaining) {
+                            if (remaining === 0) {
```

Comment:
Yeah that's what would happen. The reason this check is necessary is because of how the entire method is ordered.